wasm/types.go

```diff
@@ -22,13 +23,13 @@ type Unmarshaler interface {
 }
 
 // ValueType represents the type of a valid value in Wasm
-type ValueType int8
+type ValueType uint8
 
 const (
-	ValueTypeI32 ValueType = -0x01
-	ValueTypeI64 ValueType = -0x02
-	ValueTypeF32 ValueType = -0x03
-	ValueTypeF64 ValueType = -0x04
+	ValueTypeI32 ValueType = 0x7f
+	ValueTypeI64 ValueType = 0x7e
+	ValueTypeF32 ValueType = 0x7d
+	ValueTypeF64 ValueType = 0x7c
```

Comment:
These constants are obtained from the specification: https://webassembly.github.io/spec/core/binary/types.html#binary-valtype